jaseci_core/jaseci/utils/jsorc.py

```diff
@@ -0,0 +1,180 @@
+        cpu_usage = cpu[nodeName]
+        print(f"Detect CPU Usage: {cpu_usage}")
+        if cpu_usage > 10:
+            pods = self.controller.get_deployment_list()
+            for pod in pods:
+                namespace = pod["namespace"]
+                name = pod["name"]
+                if name == deploymentName:
+                    print("Kill deployment")
+                    self.controller.kill_deployment(name=name, namespace=namespace)
+        if cpu_usage < 5:
+            pods = self.controller.get_deployment_list()
+            redisRunning = False
+            for pod in pods:
+                if pod["name"] == deploymentName:
+                    redisRunning = True
+            if not redisRunning:
+                print("Creating new deployment")
+                self.controller.create_deployment(
+                    config=yaml.safe_load(open("jaseci.yaml", "r"))
+                )
+
+    def strategy_service_cpu(self):
+        cpu = self.promon.cpu_utilization_per_pod_cores()
+        count = 0
+        total = 0
+        for podName in cpu.keys():
+            if podName.startswith("jaseci-redis"):
+                count = count + 1
+                total = total + cpu[podName]
+
+        avg = total / count
+        print(avg)
+        if avg > 0.001:
+            conf = self.controller.get_deployment_conf("jaseci-redis", "default")
+            replicas = conf.spec.replicas + 1
+            print(f"Num of Replicas: {replicas}")
+            self.controller.deployment_set_scale("jaseci-redis", "default", replicas)
+        if avg < 0.001:
+            conf = self.controller.get_deployment_conf("jaseci-redis", "default")
+            replicas = conf.spec.replicas - 1
+            print(f"Num of Replicas: {replicas}")
+            self.controller.deployment_set_scale("jaseci-redis", "default", replicas)
+
+
+def daemon(k8sConf, prometheusURL: str):
+    m = Monitor(prometheusURL, k8sConf)
+    while True:
+        # m.strategy_redis_cpu("minikube", "default", "jaseci-redis")
+        m.strategy_start_redis()
+        time.sleep(10)
+
+
+def startMonitoring(k8sConf, prometheusURL: str):
+    monitor = multiprocessing.Process(target=daemon, args=(k8sConf, prometheusURL))
+    monitor.start()
+    return monitor
+
+
+def waitMonitoring(monitorThread):
+    monitorThread.join()
+
+
+def stopMonitoring(monitorThread):
+    monitorThread.terminate()
+
+
+def remoteK8sConf(K8sURL: str):
+    k8sconf = client.Configuration()
+    k8sconf.host = K8sURL
+    k8sconf.verify_ssl = False
+    return k8sconf
+
+
+def incluster():
+    try:
+        config.load_incluster_config()
+    except config.config_exception.ConfigException:
+        return False
+    return True
+
+
+def jsorc_start():
+    logger.info("JSORC Running")
+    if incluster():
+        k8sConfig = None
+        monitorThread = startMonitoring(
+            k8sConf=k8sConfig, prometheusURL="http://js-prometheus:9090"
+        )
```

Comment:
@Gorgeous-Patrick you can actually just give k8sconf a default value as None and then not passing anything here.